functions.lua

```diff
@@ -362,6 +433,131 @@ function M.get_crossref(str)
   return ret
 end
 
+-- function for print table values-keys recursively for debugging
+function M.printTable(t,l)
+ local nextlevel=l+1
+ for k,v in pairs(t) do
+  if type(v)=='table' then
+   M.printTable(v,nextlevel)
+  else
+   print (l..'. '..k..': '..v)
+  end
+ end
+end
+
+-- function for trying to convert (arXiv API returned) titles properly case-preserving string by BibTeX
+-- + converting greek letters and leq/geq chars in titles by their TeX commands
+function M.ucaseTitle(t)
+ local to=t:gsub('(%$[^%$]+%$)','{%1}')
+ if (t:find('%s%l%l%l%l%l')) then -- there exists at least one at least 5 character long lowercase word
+   to=to:gsub('(%u+)','{%1}')
+ end
+ local Greeks = {'alpha','beta','Gamma','gamma','Delta','delta','epsilon','varepsilon',
+  'zeta','eta','Theta','theta','vartheta','iota','kappa','Lambda','lambda',
+  'mu','nu','xi','Xi','Pi','pi','rho','varrho','Sigma','sigma','varsigma',
+  'tau','Upsilon','upsilon','Phi','phi','varphi','chi','Psi','psi','Omega','omega',
+  'leq','geq'}
+ local Grchars = {'α','β','Γ','γ','Δ','δ','ϵ','ε',
```

Comment:
I think I'd rename these two variables to `macros` and `chars` or something like this, given that they have been extended beyond the (initial?) set of only greek characters (and may be extended even further in the future)

Reply:
Thank you, I renamed the variables as you suggested.

Reply:
(see the changes of the latest commit of the PR)